lib/datasource/docker/quay.ts

```diff
@@ -0,0 +1,55 @@
+import { logger } from '../../logger';
+import { ExternalHostError } from '../../types/errors/external-host-error';
+import * as packageCache from '../../util/cache/package';
+import { http } from './common';
+
+export async function getTagsQuayRegistry(
+  repository: string
+): Promise<string[] | null> {
+  const registry = 'https://quay.io';
+  let tags: string[] = [];
+  try {
+    const cacheNamespace = 'datasource-docker-tags';
+    const cacheKey = `${registry}:${repository}`;
+    const cachedResult = await packageCache.get<string[]>(
+      cacheNamespace,
+      cacheKey
+    );
+    // istanbul ignore if
+    if (cachedResult !== undefined) {
+      return cachedResult;
+    }
```

Comment:
Move cache handling a level up to deduplicate code, you can rename the `getTags` to something like `getDockerApiTags` and add a new `getTags` which handles the quay test and the caching.